components/pedigree/resources/src/main/resources/pedigree/view/workspace.js

```diff
@@ -139,17 +140,17 @@ define([
             // set display:block
             svgText = svgText.replace(/(<svg[^<>]+style=")/g, "$1display:block; ");
```

Comment:
Maybe instead of removing elements from the pedigree to display a slimmed
down one, we should instead re-do how a pedigree is built, and the
slimmed-down pedigree would be generated by building a pedigree, and
stopping before you add all the fancy features that are removed here. Also,
RE: cloning node, take a look at [Node.cloneNode()](
https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode). The
performance is ~20% of innerHTML.

On Thu, Nov 19, 2015 at 5:51 PM, allasm <notifications@github.com> wrote:

> In
> components/pedigree/resources/src/main/resources/pedigree/view/workspace.js
> <https://github.com/phenotips/phenotips/pull/1927#discussion_r45413310>:
>
> > @@ -139,17 +140,17 @@ define([
> >              // set display:block
> >              svgText = svgText.replace(/(<svg[^<>]+style=")/g, "$1display:block; ");
>
> Good point. Unfortunately, there is a reason - the idea is to get a copy
> of the SVG, and leave the original as is. It is possible to modify DOM for
> the original, but then all of the things removed should be added back, all
> even handlers for removed parts restored, etc. This can be done, but it is
> very messy to re-run all the code which generated all of those things which
> need to be removed in the copy (@marta <https://github.com/marta> also
> think we should investigate this path, and I agree it is a good idea in
> general but it is not easy). This may also be visibly slow for a large
> pedigree (though this needs to be tested)
>
> P.S. There are some things which are also hard to do in DOM due to browser
> bugs. E.g. when a gradient is used in an SVG, Chrome (and only Chrome)
> creates a element for each gradient. And when an element is removed, that
> is not removed (which is kind of ok in itself, but makes SVG messy). But
> even worse, if a copy of an SVG is used with the same , and then the copy
> is removed, the gradients will break in the original SVG as well, even
> though original still has the - though not sure what happens when they are
> removed and then redefined anew
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phenotips/phenotips/pull/1927/files#r45413310>.
>



-- 
Denys Pavlov